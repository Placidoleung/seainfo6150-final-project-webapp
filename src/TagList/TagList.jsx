import React from 'react'
import styles from './TagList.module.css'

const TagList = (props) => {

    return(

        <div className={styles.container}>
            <h3>Tags</h3>

            <div className={styles.list}>
                {props.tags.map(tag => (
                    <div className={styles.tag} key={tag}><a href={'/tags/'+tag}>{tag}</a></div>
                ))}
            </div>

        </div>

    );

}

export default TagList;