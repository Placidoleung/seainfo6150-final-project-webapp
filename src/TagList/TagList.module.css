.tag{
    display: flex;
    align-items: center;
    margin-top: 15px;
    padding:5px;
    width:fit-content;
    background-color: orange;
    color: white;
    font-size: medium;
    font-weight: bold;
}

.tag a{
    color: white;
    text-decoration: none;
}