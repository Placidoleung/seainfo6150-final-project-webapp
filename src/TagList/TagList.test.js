import React from 'react'
import TagList from "./TagList";

describe('TagList tests', () => {

    it('render correctly ', () => {
        const tags = ["BITCOIN","DEFI"];

        const {container} = render(<TagList tags={tags}/>);
        expect(container).toMatchSnapshot();
    });

})