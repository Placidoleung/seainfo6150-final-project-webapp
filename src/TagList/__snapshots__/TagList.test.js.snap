// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList tests render correctly  1`] = `
<div>
  <div
    class="container"
  >
    <h3>
      Tags
    </h3>
    <div
      class="list"
    >
      <div
        class="tag"
      >
        <a
          href="/tags/BITCOIN"
        >
          BITCOIN
        </a>
      </div>
      <div
        class="tag"
      >
        <a
          href="/tags/DEFI"
        >
          DEFI
        </a>
      </div>
    </div>
  </div>
</div>
`;
